Helper/AutomatedTesting.php

```diff
@@ -281,6 +380,70 @@ protected function convertToStoreItem($product, $type)
             ->setType($type);
     }
 
+    /**
+     * Convert $address to an Address
+     *
+     * @param ModelOrderAddress|null $address
+     *
+     * @return Address|null
+     */
+    protected function convertToAddress($address)
+    {
+        if ($address === null) {
+            return null;
+        }
+        return $this->addressFactory->create()
+            ->setFirstName($address->getFirstName())
+            ->setLastName($address->getLastName())
+            ->setStreet(implode(" ", $address->getStreet()))
+            ->setCity($address->getCity())
+            ->setRegion($address->getRegion())
+            ->setPostalCode($address->getPostCode())
+            ->setTelephone($address->getTelephone())
+            ->setCountry($address->getCountryId());
+    }
+
+    /**
+     * Convert $order to an Order
+     *
+     * @param ModelOrder $order
+     *
+     * @return Order
+     */
+    protected function convertToOrder($order)
+    {
+        if ($order === null) {
+            return null;
+        }
+        $orderItems = [];
+        foreach($order->getAllVisibleItems() as $item) {
+            $orderItems[] = $this->orderItemFactory->create()
+                ->setProductName($item->getName())
+                ->setProductSku($item->getSku())
+                ->setProductUrl($item->getProduct()->getProductUrl())
+                ->setPrice($this->formatPrice($item->getPrice()))
+                ->setQuantityOrdered($item->getQtyOrdered())
+                ->setSubtotal($this->formatPrice($item->getRowTotal()))
+                ->setTaxAmount($this->formatPrice($item->getTaxAmount()))
+                ->setTaxPercent($item->getTaxPercent())
+                ->setDiscountAmount($this->formatPrice($item->getDiscountAmount()))
+                ->setTotal($this->formatPrice($item->getRowTotal()
+                    + $item->getTaxAmount()
+                    - $item->getDiscountAmount()));
```

Comment:
```suggestion
                ->setTotal($this->formatPrice(
                    $item->getRowTotal() +
                    $item->getTaxAmount() -
                    $item->getDiscountAmount()
                ));
```

Reply:
so discount_amount is negative, but getDiscountAmount is positive? just to confirm

Reply:
done